Add better autoconf support for libboost

dongahn opened the pull request:
Add `powerpc64le` and `armv7l` platforms to `ax_boost_base.m4`
such that configure will search for the system library path where
boost RPMs install their boost libraries.

Add the actual boost library names in the config error messages
(`ax_boost_filesystem.m4`, `ax_boost_system.m4`
and `ax_boost_graph.m4`).

Add a new m4 script that checks whether the buggy libboost versions
as tested and documented at #297. 

Resolve #302,  #297 and #396.

SteVwonder commented:
Ouch.  When I updated the dependencies in the (now old) dep builder, I probably broke something.  Sorry!  I agree that merging #392 should fix this.

garlick commented:
There's a missing comma in the armv7l patch, but after I add that, configure seems to find all the boost bits!
```
checking for boostlib >= 1.53.0 (105300)... yes
checking for buggy boostlib versions (105400-105800)... no
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
checking for exit in -lboost_filesystem... yes
checking whether the Boost::Graph library is available... yes
checking for exit in -lboost_graph... yes
checking whether the Boost::Regex library is available... yes
checking for exit in -lboost_regex... yes
```

```patch
diff --git a/m4/ax_boost_base.m4 b/m4/ax_boost_base.m4
index f750fa4..c9749dc 100644
--- a/m4/ax_boost_base.m4
+++ b/m4/ax_boost_base.m4
@@ -114,7 +114,7 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
     AS_CASE([${host_cpu}],
       [x86_64],[libsubdirs="lib64 libx32 lib lib64"],
       [ppc64|s390x|sparc64|aarch64|ppc64le|powerpc64le],[libsubdirs="lib64 lib lib64"],
-      [armv7l],[libsubdirs="lib/arm-linux-gnueabihf lib"]
+      [armv7l],[libsubdirs="lib/arm-linux-gnueabihf lib"],
       [libsubdirs="lib"]
     )

```

dongahn commented:
Great! I didn't have a test machine and that crept in. Sorry about that.

garlick commented:
No worries!

SteVwonder commented on configure.ac:
```diff
@@ -62,7 +62,9 @@ AC_DEFINE_UNQUOTED([LIBLUA_SO], ["$LIBLUA_SONAME"],
 PKG_CHECK_MODULES([UUID], [uuid], [], [])
 AX_FLUX_CORE
+
 AX_BOOST_BASE([1.53.0], [], [AC_MSG_ERROR([Please install boost >= 1.53)])])
```
Can you update this error message to match the one in buggy boost? `Please use Boost == 1.53 or > 1.58`

SteVwonder commented:
Other than the one nit, LGTM.  @dongahn, maybe you can take a look at #396 and see if the boost portion of that makes sense to include in this PR.  If not, I'm happy to merge this PR without it.

dongahn commented:
@SteVwonder and @garlick: I addressed both of the issues you found. Also, as part of that, I addressed the issues captured in #396 as well. Thanks. 

@SteVwonder: everything turned green and from my perspective this can go in.

SteVwonder commented:
Thanks @dongahn! Merging.
